### DISCUSSION TITLE
Generate dmp File out modified JSON?

### DISCUSSION THREAD

**Discussion by RaW79** [tasmota/decode-config#89] [Q&A]
Hi

my intention is to have a configuration file of my Tasmota devices which is easy/standard to recover -> dmp file, but without Wifi credentials in it.

So I used decode-config to generate readable configuration -> JSON and deleted the Wifi credentials in it.

Can you please explain and help how to create a dmp file out of this JSON file?

I could not achieve it so far:

![grafik](https://github.com/tasmota/decode-config/assets/16465702/e52bf1d4-4453-4862-8845-79bdff35a255)

Thanks

**Comment by curzon01** [comment]
> Can you please explain and help how to create a dmp file out of this JSON file?

A source can only be an **existing** one on the left side from [this picture](https://github.com/tasmota/decode-config?tab=readme-ov-file#decode-config). Source can only be a physical device via http or mqtt or a dmp file but not a json and you can't create a none existing dmp. Source must always exists. Pls first have a look on the main readme first picture left side (Source) and right side (Output).

You can overwrite the content of an existing dmp file with the json data but keep in mind, a dmp file always contains **all** config data, you can not use only parts of them, Tasmota does not allow partly use of dmp data. This was also one reason why I wrote decode-config, to handle part of config using json.

See also #53 and https://github.com/tasmota/decode-config/discussions/77#discussioncomment-8451546

So if you would exclude parts you have to use a json as restore file and the physical device as source.
I'm wondering why you want tonuse this extra step, creating a dmp from a json, then load this dmp into your Tasmota device. Simply use the device directly using your modified json

`decode-config --source tasmota-device --restore-file 1234.json`
